app/src/main/java/com/alexstyl/themingandroid/AppCompatThemedActivity.java

```diff
@@ -38,10 +38,10 @@ protected final void applyTheme(AppTheme theme) {
     }
 
     private void restartActivity() {
-        Intent intent = getIntent();
+        final Intent intent = getIntent();
         intent.removeCategory(Intent.CATEGORY_LAUNCHER);
-        startActivity(intent);
         finish();
+        startActivity(intent);
```

Comment:
But the sample contains no fragments. Am I missing something?

Starting the activity after finishing the previous one modifies behavior. The pending animation is overwritten to be a fade. If you finish the current activity first, then you should have a short period of 'emptiness' appearing on screen. 
Starting the activity first ensures that there is always something on screen while the animation is playing.